pkg/info/info_test.go

```diff
@@ -0,0 +1,40 @@
+package info
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("eksctl info test", func() {
+
+	It("ignores kubectl version and OS", func() {
+		info := GetVersion()
+
+		Expect(info).To(Equal(Version{
+			EksctlVersion:  "0.45.0",
+			KubectlVersion: "",
+			OS:             "",
+		}))
+	})
+
+	It("ignores OS", func() {
+		info := GetVersion()
+
+		Expect(info).To(Equal(Version{
+			EksctlVersion:  "0.45.0",
+			KubectlVersion: "v1.18.1",
+			OS:             "",
+		}))
+	})
+
+	It("wrong eksctl, kubectl versions and OS", func() {
+		info := GetVersion()
+
+		Expect(info).To(Equal(Version{
+			EksctlVersion:  "0.40.0",
+			KubectlVersion: "v1.10.1",
+			OS:             "amd64",
+		}))
+	})
+
```

Comment:
I think this test can be simpler. The `kubectl` and `OS` are very unlikely to be ignored, although I am sure you can find a way to force them to disappear. Likewise there will never be a "wrong" version for any of them.

```suggestion
 var _ = Describe("Info", func() {
   Context("GetInfo", func() {
     It("returns the eksctl and kubectl versions, and the host os", func() {
       result := info.GetInfo()

       Expect(result.EksctlVersion).ToNot(Equal(""))
       Expect(result.KubectlVersion).ToNot(Equal(""))
       // we can't check for a specific version since it will change depending
       // on who is running the tests, but we can check that actual semvers are returned
       _, err := semver.NewVersion(strings.TrimRight(result.EksctlVersion, "\n"))
       Expect(err).NotTo(HaveOccurred())
       _, err = semver.NewVersion(strings.TrimRight(result.KubectlVersion, "\n"))
       Expect(err).NotTo(HaveOccurred())

       Expect(result.OS).ToNot(Equal(""))
       // technically a cheat, but
       // testing for the OS is harder, we could possibly have an array of OSes and see
       // if the one here  matches, if you feel like doing that
       Expect(result.OS).To(Equal(runtime.GOOS))
     })
   })

   Context("String", func() {
     It("returns Info in json", func() {
       result := info.String()

       // make processing easier, turn into a struct
       versions := info.Versions{}
       err := json.Unmarshal([]byte(result), &versions)
       Expect(err).NotTo(HaveOccurred())

       // same again basically, we can just check they are not nil
       Expect(versions.EksctlVersion).ToNot(Equal(""))
       Expect(versions.KubectlVersion).ToNot(Equal(""))
       Expect(versions.OS).ToNot(Equal(""))
     })
   })
   
   // There are is a potential failure case, but it is trickier
   // to test at this level. To test that the right thing happens when kubectl fails
   // to execute, you could remove the binary at the start of the test and put it back at the end
   // but if that goes wrong then you dont have a kubectl anymore :D 
   // Another option is to have the kubectl binary be an exported location which you can reset
   // in the test to a dummy bin. But this is a lot of work for a small functionality,
   // so if that sound like a pain I am happy to ignore it 
 })
```

Reply:
@Callisto13 thank you so much! Those tests definitely simplified everything.
1. regarding the OS test, it looks like these OSes are supported: https://github.com/golang/go/blob/master/src/go/build/syslist.go#L10. 
2. The kubectl test of moving the binary sounds a bit of an overkill. 

I feel the tests you have suggested look great. And thanks again for helping me out.

Reply:
> regarding the OS test, it looks like these OSes are supported: https://github.com/golang/go/blob/master/src/go/build/syslist.go#L10.

cool, if you want to implement a match loop there feel free. 

> The kubectl test of moving the binary sounds a bit extra.

👍  no worries